exec/exec.go

```diff
@@ -0,0 +1,29 @@
+// Package exec adapts the lego exec DNS
+// provider for Caddy. Importing this package plugs it in.
+package exec
+
+import (
+	"errors"
+
+	"github.com/mholt/caddy/caddytls"
+	"github.com/xenolf/lego/providers/dns/exec"
+)
+
+func init() {
+	caddytls.RegisterDNSProvider("exec", NewDNSProvider)
+}
+
+// NewDNSProvider returns a new exec DNS challenge provider.
+// The credentials are interpreted as follows:
+//
+// len(0): use credentials from environment
+// len(2): credentials[0] = Email address
+//         credentials[1] = API key
+func NewDNSProvider(credentials ...string) (caddytls.ChallengeProvider, error) {
+	switch len(credentials) {
+	case 0:
+		return exec.NewDNSProvider()
```

Comment:
There ought to be a way to pass in the command by argument if desired, no?

Reply:
Well, you tell me. I didn't see any obvious way for a DNS provider to get config settings from the Caddyfile. If it existed I guess that would be somewhere here: https://github.com/mholt/caddy/blob/master/caddytls/client.go#L198

Reply:
I have put in a PR for lego to allow the program to be set directly instead of via environment variable: https://github.com/xenolf/lego/pull/573
If that gets merged, I will add a `case 1:` here, that allows to set the command directly. This way, if Caddy gets the ability to pass in config to providers, it would automatically work with the exec provider.

Reply:
:+1: Perfect - thanks!